lib/matplotlib/dates.py

```diff
@@ -282,6 +283,11 @@ def _from_ordinalf(x, tz=None):
         tz = _get_rc_timezone()
 
     ix = int(x)
+    if ix < 1:
+        raise ValueError('cannot convert {} < 1 to a date.  This '
+                         'often happens if floats are passed to '
+                         'a Matplotlib axis that expects datetime objects. '
+                         ''.format(ix))
```

Comment:
you don't need the `''`, starting the line with a dot binds to the string on the previous line